idunn/blocks/wikipedia.py

```diff
@@ -199,14 +198,13 @@ def with_cache(*args, **kwargs):
             otherwise we bypass it
             """
             if cls._connection is not False:
-                value_stored = cls.get_value(key, f)
+                value_stored = cls.get_value(key)
                 if value_stored is not None:
-                    result = json.loads(value_stored.decode('utf-8'))
-                else:
-                    result = f(*args, **kwargs)
-                    json_result = json.dumps(result)
-                    if not cls.set_value(f, key, json_result):
-                        return None
+                    return json.loads(value_stored.decode('utf-8'))
+                result = f(*args, **kwargs)
+                json_result = json.dumps(result)
+                if not cls.set_value(key, json_result):
+                    return None
```

Comment:
if we can't write the response in the cache, won't we cant to return this response anyway ?

Reply:
indeed you're right,
done